.gitlab-ci-check-docker-build.yml

```diff
@@ -241,6 +241,47 @@ publish:image:mender:
     reports:
       dotenv: publish.env
 
+publish:image-multiplatform:mender:
+  tags:
+    - mender-qa-worker-generic-light
+  stage: publish
+  rules:
+    - if: $MULTIPLATFORM_BUILD != "true"
+      when: never
+    - if: '$CI_COMMIT_BRANCH =~ /^(master|[0-9]+\.[0-9]+\.x)$/'
+  image: "registry.gitlab.com/northern.tech/mender/mender-test-containers:docker-multiplatform-buildx-v1-master"
+  services:
+    - docker:20.10.21-dind
+  dependencies:
+    - build:docker-multiplatform
+  before_script:
+    # Use same variables for loading the image, while DOCKER_PUBLISH_COMMIT_TAG will be ignored
+    - docker login -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY
+    - *export_docker_vars
+    - *get_release_tool_alpine
+    - *docker_login_registries
+  script:
+    # If the repo is not recognized, ignore
+    - if ! echo $(release_tool --list git --all) | grep $CI_PROJECT_NAME; then
+    -  echo "Repository $CI_PROJECT_NAME not found in release_tool. Exiting"
+    -  exit 0
+    - fi
+    # If the repo/branch is not part of a Mender release, also ignore
+    - integration_versions=$(release_tool --integration-versions-including $CI_PROJECT_NAME --version $CI_COMMIT_REF_NAME | sed -e 's/origin\///')
+    - if test -z "$integration_versions"; then
+    -  echo "Repository $CI_PROJECT_NAME version $CI_COMMIT_REF_NAME is not part of any Mender release. Exiting"
+    -  exit 0
+    - fi
+    # Publish the image for all releases
+    - for version in $integration_versions; do
+    -   regctl image copy ${CI_REGISTRY_IMAGE}:${CI_PIPELINE_ID} $DOCKER_REPOSITORY:mender-${version}
+    -   regctl image copy ${CI_REGISTRY_IMAGE}:${CI_PIPELINE_ID} $DOCKER_REPOSITORY:mender-${version}_${CI_COMMIT_SHA}
+    - done
+    - echo "PUBLISH_IMAGE_DIGEST=${DOCKER_REPOSITORY}@$(regctl image digest $DOCKER_REPOSITORY:mender-${version})" >> publish.env
```

Comment:
I didn't know about `regctl` (thanks!) which seems to be very handy. Could we port the use of the tool to `publish:image:mender:` and `publish:image:`? It probably means also changing the base image for the CI job.

Reply:
Yes, `regctl` (or other alternative tools) must be used because the `docker tag` itself cannot deal with multiplatform images.
We could port it also to the non-multiplatform jobs (`publish:image:*`), maybe let's deal it in a separate ticket